### PR TITLE
feat(@spartacus/core): Move cxApi to core 

### LINKED ISSUE #951: Move cxApi to core

### PR DESCRIPTION
**Pull request by Pucek9**
Move cxApi to core 

Closes #951

### CODE REVIEW

File: projects/core/src/cx-api/cx-api.service.spec.ts [hunk 1]
```diff
@@ -1,16 +1,15 @@
 import { TestBed, inject } from '@angular/core/testing';
 import { CxApiService } from './cx-api.service';
 
+import { AuthService } from '../auth';
```
Comment (kacperknapik) [inline projects/core/src/cx-api/cx-api.service.spec.ts]: each import should use the corresponding barrels - if the barrel exists (lines 4-12)

File: projects/core/src/cx-api/cx-api.service.ts [hunk 2]
```diff
@@ -1,21 +1,15 @@
-import { Injectable, Optional } from '@angular/core';
+import { Optional } from '@angular/core';
+import { AuthService } from '../auth';
```
Comment (kacperknapik) [inline projects/core/src/cx-api/cx-api.service.ts]: each import should use the corresponding barrels - if the barrel exists (lines 2-11)

File: projects/storefrontlib/src/lib/storefront.module.ts [hunk 3]
```diff
@@ -13,7 +13,6 @@ import { StorefrontModuleConfig } from './storefront-config';
 
 import { CmsLibModule } from './cms-lib/index';
 import { CmsModule } from './cms/index';
-import { CxApiModule } from './cx-api/cx-api.module';
```
Comment (kacperknapik) [inline projects/storefrontlib/src/lib/storefront.module.ts]: aren't we using this module in the storefront?

Comment (kacperknapik) [reply projects/storefrontlib/src/lib/storefront.module.ts]: if so, please leave this import and the usage below, if not just ignore this comment

Comment (Pucek9) [reply projects/storefrontlib/src/lib/storefront.module.ts]: That's module I moved from storefront to core :) It doesn't exist anymore there

**Pull request merged**